Adding documentation for Locale (#16)
gr4vy/gr4vy-docs: guides/embed/locale.mdx

---
title: Locale
slug: /embed/locale
---

import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";
import Image from "../../src/modules/Image";
import image from "./assets/embed-theme.png";

## Overview

Embed can be configured to use a different language using the `locale` property. By default, this will be set to `en` (English).
Embed accepts a [ISO 639-1 Language Code](https://www.iso.org/iso-639-language-codes.html) (e.g. `en` or `fr`) and an optional [ISO 3166 Country Code](https://www.iso.org/iso-3166-country-codes.html) (e.g. `GB` or `NL`).
The country code allows us to provide a localized version of the language based on a specific country, e.g. `fr-CA` would represent Canadian French.

## Usage

> A locale from the list of [Supported Locales](#supported-locales) can be set.

Reviewer: ```suggestion
The locale can be set to any of the [supported Locales](#supported-locales).
```

> The locale property will fallback to English if a locale is not provided, is invalid or not supported.

Reviewer: ```suggestion
The locale property will fall back to English if a locale is not provided, is incorrect or not supported.
```

> **Please Note: Embed currently does not try to detect the buyers locale automatically**.

Reviewer: ```suggestion
> Note: Embed currently does not automatically detect the buyer's locale. The
> locale will always be set to `en` when not provided.
```

<Tabs
  groupId="language"
  defaultValue="react"
  values={[
    { label: 'React', value: 'react', },
    { label: 'CDN', value: 'cdn', },
  ]}
>
  <TabItem value="react">

```tsx
<Gr4vyEmbed currency="GBP" amount="2300" locale="en-GB" />
```

  </TabItem>

  <TabItem value="cdn">

```js
setup({
  ...,
  amount: 1299,
  country='GB'
  currency: "GBP",
  locale: 'en-GB'
});
/>
```

  </TabItem>
</Tabs>

## Supported Locales

We currently support the following locales.

| Locale Name          | Language Code (ISO 639-1) | Country Code (ISO 3166) | Locale |
| -------------------- | ------------------------- | ----------------------- | ------ |
| American English     | en                        | US                      | en-US  |
| British English      | en                        | GB                      | en-GB  |
| Brazilian Portuguese | pt                        | BR                      | pt-BR  |
| French               | fr                        |                         | fr     |
| Spanish              | es                        |                         | es     |

> | Portuguese           | pt                        |                         | pt     |

Reviewer: I'd put all these codes in backticks or the spellchecker will probably fail.

## FAQ

### Will more locales be supported?

> We currently offer a limited set of locales and plan to roll-out more support in the future, including right-to-left languages such as Arabic.

Reviewer: I'd drop this until we fully make this decision.
